### PR TITLE
[RF-6179] Fix sc4 executable on Windows

### PR DESCRIPTION
**Pull request by magni-**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,6 +167,36 @@ results.each do |test_id, browsers|
...
+
+### Windows
+
+There is an [open issue](https://github.com/drbrain/net-http-persistent/issues/79) in a dependency of a dependency, which causes the following error on Windows:
+```
+NameError: uninitialized constant Process::RLIMIT_NOFILE
+  from (irb):1
+```
+
+Until a fix is released to that gem, the tempfix is to manually edit that gem's source:
```
Comment (grodowski) [inline README.md]: I would consider monkey patching this in our gem, would that work?

```
Net::Http::Persistent::DEFAULT_POOL_SIZE = ...
```

File: Rakefile [hunk 2]
```diff
@@ -11,17 +12,20 @@ RSpec::Core::RakeTask.new(:spec) do |t|
...
-    case RUBY_PLATFORM
-    when /cygwin|mswin|mingw|bccwin|wince|emx/ then 'sc-windows.exe'
-    when /darwin/ then 'sc-osx'
+    case platform
+    when :windows then 'sc-windows.exe'
+    when :osx then 'sc-osx'
     else 'sc-linux'
     end
 
+  extension = platform == :windows ? '.exe' : ''
```
Comment (grodowski) [inline Rakefile]: Is it worth DRYing this up (with `Drivers::Sauce#apply_config` L25)? Could be a util method `sc_executable_name` on the main module.

File: lib/rainforest_ruby_runtime/drivers/sauce.rb [hunk 3]
```diff
@@ -22,9 +22,12 @@ def to_rspec(tests)
       private
 
       def apply_config
+        sc4_executable = RainforestRubyRuntime.platform == :windows ? 'sc.exe' : 'sc'
+
         ::Sauce.config do |c|
           c[:browsers] = browsers
-          c[:sauce_connect_4_executable] = File.join(RainforestRubyRuntime.root, '/vendor/sauce/sc')
+          c[:sauce_connect_4_executable] = File.join(RainforestRubyRuntime.root, '/vendor/sauce/', sc4_executable)
+          c[:start_local_application] = false
```
Comment (grodowski) [inline lib/rainforest_ruby_runtime/drivers/sauce.rb]: 😇

File: README.md (hunk 1 above)
Comment (magni-) [reply README.md]: Couldn't get it to work...

**Review by grodowski — APPROVED** [review]

**Pull request merged**